scipy/optimize/_optimize.py

```diff
@@ -137,14 +137,11 @@ def _wrap_callback(callback, method=None):
     if callback is None or method in {'tnc', 'slsqp', 'cobyla'}:
         return callback  # don't wrap
 
-    p1 = (inspect.Parameter('intermediate_result',
-                            kind=inspect.Parameter.POSITIONAL_OR_KEYWORD),)
-    p2 = (inspect.Parameter('intermediate_result',
-                            kind=inspect.Parameter.KEYWORD_ONLY),)
-    sig1 = inspect.Signature(parameters=p1)
-    sig2 = inspect.Signature(parameters=p2)
-
-    if inspect.signature(callback) in {sig1, sig2}:
+    sig = inspect.signature(callback)
+    has_one_parameter = len(sig.parameters) == 1
+    named_intermediate_result = sig.parameters.get('intermediate_result', 0)
```

Comment:
Works although 0, 1 for bool is not very clear to me. Personally I tend to forget which is which. Why not being more specific with:
```suggestion
    named_intermediate_result = sig.parameters.get(
        'intermediate_result', False
    )
```

Reply:
Yeah I was being sloppy. I didn't want to split lines, but I should have just changed the variable name.
But the other suggestion resolved this, too